mono/profiler/log.c

```diff
@@ -512,6 +518,8 @@ init_thread (gboolean add_to_lls)
 {
 	MonoProfilerThread *thread = PROF_TLS_GET ();
 
+	g_assert (thread != MONO_PROFILER_THREAD_DEAD && "Why are we trying to resurrect a stopped thread?");
```

Comment:
If this keep happening frequently, we could return a dummy MonoProfilerThread instead.

Reply:
It should never happen, and indeed never did outside #5710, where it only happened because it introduced events that got fired after the `thread_stopped` event. I think an assertion is fine, we don't want a bug like this to go unnoticed. (Also, it already passed the stress tests in an earlier run of this PR, so I think we're good.)